node/impl/full/actor_events.go

```diff
@@ -321,30 +338,30 @@ func (a *ActorEventHandler) SubscribeActorEvents(ctx context.Context, evtFilter
 				case <-ctx.Done():
 					return
 				case <-ticker.C:
-					if params.MaxHeight > 0 && a.Chain.GetHeaviestTipSet().Height() > params.MaxHeight {
+					currentHeight := a.chain.GetHeaviestTipSet().Height()
+					if params.MaxHeight > 0 && currentHeight > params.MaxHeight {
+						// we've reached the filter's MaxHeight, we're done so we can close the channel
 						return
 					}
 				}
 			}
+
+			if a.clock.Now().After(nextBacklogHeightUpdate) {
+				minBacklogHeight = a.chain.GetHeaviestTipSet().Height() - 1
+				nextBacklogHeightUpdate = a.clock.Now().Add(a.blockDelay)
```

Comment:
I wouldn't bother with time here. Just wait till we see a new event, and check the height.

Reply:
Ah, I see what you mean...